src/integrations/iron_handlers.rs

```diff
@@ -7,11 +7,19 @@ use iron::status;
 use iron::method;
 use iron::url::Url;
 
-use std::collections::BTreeMap;
+use std::io::Read;
+use std::io::Error as IoError;
+use std::io::ErrorKind;
+use std::collections::HashMap;
+use std::error::Error;
+use std::fmt;
+use std::boxed::Box;
 
-use rustc_serialize::json::{ToJson, Json};
+use serde_json;
+use serde_json::Value as Json;
+use serde_json::error::Error as SerdeError;
 
-use ::{InputValue, GraphQLType, RootNode, Variables, execute};
+use ::{InputValue, GraphQLType, RootNode, Variables, execute as execute_query};
```

Comment:
Changed the name of this in order to clarify the difference between the execute function that exists in the iron_handlers file vs the execute function that resides in the lib.rs file.